source/acquaman/AMGenericStepScanConfiguration.cpp

```diff
@@ -0,0 +1,187 @@
+	configuration->dissociateFromDb(true);
+	return configuration;
+}
+
+AMScanController *AMGenericStepScanConfiguration::createController()
+{
+	AMScanActionController *controller = new AMGenericStepScanController(this);
+	controller->buildScanController();
+
+	return controller;
+}
+
+AMScanConfigurationView *AMGenericStepScanConfiguration::createView()
+{
+	return new AMGenericStepScanConfigurationView(this);
+}
+
+QString AMGenericStepScanConfiguration::technique() const
+{
+	return "Generic Scan";
+}
+
+QString AMGenericStepScanConfiguration::description() const
+{
+	return "Generic Scan";
+}
+
+QString AMGenericStepScanConfiguration::detailedDescription() const
+{
+	return "Does a generic scan over one or two controls with a variety of detectors.";
+}
+
+void AMGenericStepScanConfiguration::computeTotalTime()
+{
+	if (scanAxes_.count() == 1){
+
+		double size = fabs(double(scanAxes_.at(0)->regionAt(0)->regionEnd())-double(scanAxes_.at(0)->regionAt(0)->regionStart()));
+		int points = int((size)/double(scanAxes_.at(0)->regionAt(0)->regionStep()));
+
+		if ((size - (points + 0.01)*double(scanAxes_.at(0)->regionAt(0)->regionStep())) < 0)
+			points += 1;
+
+		else
+			points += 2;
+
+	}
+
+	else if (scanAxes_.count() == 2){
+
+		double hSize = fabs(double(scanAxes_.at(0)->regionAt(0)->regionEnd())-double(scanAxes_.at(0)->regionAt(0)->regionStart()));
+		double vSize = fabs(double(scanAxes_.at(1)->regionAt(0)->regionEnd())-double(scanAxes_.at(1)->regionAt(0)->regionStart()));
+
+		int hPoints = int((hSize)/double(scanAxes_.at(0)->regionAt(0)->regionStep()));
+		if ((hSize - (hPoints + 0.01)*double(scanAxes_.at(0)->regionAt(0)->regionStep())) < 0)
+			hPoints += 1;
+		else
+			hPoints += 2;
+
+		int vPoints = int((vSize)/double(scanAxes_.at(1)->regionAt(0)->regionStep()));
+		if ((vSize - (vPoints + 0.01)*double(scanAxes_.at(1)->regionAt(0)->regionStep())) < 0)
+			vPoints += 1;
+		else
+			vPoints += 2;
+
+		totalTime_ = hPoints*vPoints*double(scanAxes_.at(0)->regionAt(0)->regionTime());
+	}
+
+	setExpectedDuration(totalTime_);
+	emit totalTimeChanged(totalTime_);
+}
+
+void AMGenericStepScanConfiguration::setControl(int axisId, AMControlInfo newInfo)
+{
+	if (axisId == 0 && controls_.isEmpty()){
+
+		controls_.append(newInfo);
+
+		AMScanAxisRegion *region = new AMScanAxisRegion;
+		AMScanAxis *axis = new AMScanAxis(AMScanAxis::StepAxis, region);
+		appendScanAxis(axis);
+
+		connect(scanAxisAt(0)->regionAt(0), SIGNAL(regionStartChanged(AMNumber)), this, SLOT(computeTotalTime()));
+		connect(scanAxisAt(0)->regionAt(0), SIGNAL(regionStepChanged(AMNumber)), this, SLOT(computeTotalTime()));
+		connect(scanAxisAt(0)->regionAt(0), SIGNAL(regionEndChanged(AMNumber)), this, SLOT(computeTotalTime()));
+		connect(scanAxisAt(0)->regionAt(0), SIGNAL(regionTimeChanged(AMNumber)), this, SLOT(computeTotalTime()));
+	}
+
+	else if (axisId == 0){
+
+		controls_.replace(0, newInfo);
```

Comment:
No, we don't need to replace the controls because the controlInfo's are not being connected to anything.  All the signals are connected through the scan axes.